### PR TITLE
 [W5.6a][W13-2]Arjo Chakravarty

### PR DESCRIPTION
**Pull request by arjo129**
Extracted out a contacts class. Hopefully did this correctly

### CODE REVIEW

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * A class containing the implementations of Contact
+ */
+public class Contact {
+    public final String value;
+    protected boolean isPrivate;
+    //private final String trimmedValue;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:11]: Remove unused lines; you can retrieve them from the VCS if necessary

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    public  Contact(String value, Boolean isPrivate)  {
+        this.isPrivate = isPrivate;
+        this.value = value.trim();
+    }
+    @Override
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:16]: Please add vertical spacing for readability

File: src/seedu/addressbook/data/person/Email.java [hunk 3]
```diff
@@ -6,28 +6,23 @@
...
     /**
      * Validates given email.
      *
      * @throws IllegalValueException if given email address string is invalid.
      */
     public Email(String email, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        String trimmedEmail = email.trim();
-        if (!isValidEmail(trimmedEmail)) {
+        super(email,isPrivate);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Email.java:22]: By doing this you are assigning the value before validation.

File: test/java/seedu/addressbook/commands/AddCommandTest.java [hunk 4]
```diff
@@ -103,7 +103,7 @@ public void addCommand_validData_correctlyConstructed() throws Exception {
 
         // TODO: add comparison of tags to person.equals and equality methods to
         // individual fields that compare privacy to simplify this
-        assertEquals(Name.EXAMPLE, p.getName().fullName);
+        assertEquals(Name.EXAMPLE, p.getName().value);
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/AddCommandTest.java:106]: 👍 for updating the tests

**Review by okkhoy — APPROVED** [review]
Good job! 👍

**Comment by okkhoy** [comment]
@arjo129 
Some comments added. Please take note and close the pull request.

**Comment by arjo129** [comment]
Thank you!

**Pull request closed without merging**